dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -186,10 +186,12 @@ def _determine_pf_parts(fs, paths, gather_statistics, ignore_metadata_file, **kw
 
         elif gather_statistics is not False:
             # Scan every file
+            paths = [path for path in paths if path.endswith((".parq", ".parquet"))]
```

Comment:
Do we really want to require a specific file suffix for data files?  I know pyarrow ignores files that begin with `"_"`, but I don't think there are any limitations on the data file names.

Reply:
This is the reason, then, that is was failing for arrow - the original spark dataset contains . parq.crc files ,which are not zero in size, but not data.

Reply:
Okay - Then I guess it does make sense to filter by the extension. However, we will probably need to allow the user to modify the allowed extensions in case they have been using Dask to process data with some other convention?

Reply:
I have never come across any others!

Reply:
>I have never come across any others!

I have seen several cases where there was no extension in the sting path at all :/

Reply:
Hm, that's a problem. These extensions are what fastparquet explicitly looks for right now (although metadata files are much more common, of course). Otherwise, it would come down to *excluding* certain file names, which is harder.

Reply:
It looks like the challenge here will definitly be on the pyarrow side. Right now, the [pyarrow dataset API](https://arrow.apache.org/docs/python/generated/pyarrow.dataset.dataset.html#pyarrow.dataset.dataset) only allows you to exclude specific prefixes (no option to allow nor exclude specific suffixes). Therefore, we would need to perform file discovery outselves in Dask, and then pass the filtered list into pyarrow.

Reply:
So, do you think I should change the test to not have the crc file, so things pass (and accept that pyarrow would fail for the original spark data), just skip pyarrow for the test, or do something else?